### ISSUE TITLE
Xcode 8 and ENABLE_BITCODE error when building using Carthage

### ISSUE THREAD

**Issue by notapplicableio** [rheinfabrik/Dobby#34]
Can anyone help with this?
- Xcode 8
- Carthage

When I do an update it falls over when building Dobby. Other projects are fine.

**\* Building scheme "Dobby-iOS" in Dobby.xcworkspace
*\* BUILD FAILED **

The following build commands failed:
    Ld /Users/scott/Library/Developer/Xcode/DerivedData/Dobby-ecuwfbyesepmhtfviqolgxllubry/Build/Intermediates/Dobby.build/Release-iphoneos/Dobby-iOS.build/Objects-normal/arm64/Dobby normal arm64
    Ld /Users/scott/Library/Developer/Xcode/DerivedData/Dobby-ecuwfbyesepmhtfviqolgxllubry/Build/Intermediates/Dobby.build/Release-iphoneos/Dobby-iOS.build/Objects-normal/armv7/Dobby normal armv7

ld: warning: Auto-Linking supplied '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks/XCTest.framework/XCTest.tbd', bitcode bundle could not be generated because '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks/XCTest.framework/XCTest.tbd' was built without full bitcode. All frameworks and dylibs for bitcode must be generated from Xcode Archive or Install build
ld: warning: Auto-Linking supplied '/Applications/Xcode.app/Contents/Developer/Toolchains/Swift_2.3.xctoolchain/usr/lib/swift/iphoneos/libswiftXCTest.dylib', '/Applications/Xcode.app/Contents/Developer/Toolchains/Swift_2.3.xctoolchain/usr/lib/swift/iphoneos/libswiftXCTest.dylib' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target.
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ld: warning: Auto-Linking supplied '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks/XCTest.framework/XCTest.tbd', bitcode bundle could not be generated because '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/Library/Frameworks/XCTest.framework/XCTest.tbd' was built without full bitcode. All frameworks and dylibs for bitcode must be generated from Xcode Archive or Install build
ld: warning: Auto-Linking supplied '/Applications/Xcode.app/Contents/Developer/Toolchains/Swift_2.3.xctoolchain/usr/lib/swift/iphoneos/libswiftXCTest.dylib', '/Applications/Xcode.app/Contents/Developer/Toolchains/Swift_2.3.xctoolchain/usr/lib/swift/iphoneos/libswiftXCTest.dylib' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target.
clang: error: linker command failed with exit code 1 (use -v to see invocation)
A shell task (/usr/bin/xcrun xcodebuild -workspace /Volumes/Mobi/Mobi2GoInStoreManager/Carthage/Checkouts/Dobby/Dobby.xcworkspace -scheme Dobby-iOS -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
*\* BUILD FAILED **

**Comment by SusannProszak** [comment]
hello @trustyfrog could you please provide us a sample project with the error your getting. Thanks :)

**Comment by mdarnall** [comment]
I experienced this issue a few weeks back, before our team had converted over to Swift 3 and Dobby 0.7.0. Today I went back to see if we still had an issue using Dobby as a framework. Now that our project is Swift 3 and Dobby 0.7.0 It worked aside from the following warning (which I can file a different issue for): 

```
ld: warning: linking against a dylib which is not safe for use in application extensions: /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks/XCTest.framework/XCTest.tbd
ld: warning: linking against a dylib which is not safe for use in application extensions: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator/libswiftXCTest.dylib
```


For curiosity sake I wanted to see if the original issue here was gone. It does seem to still be an issue for users of Swift 2.3 and Dobby 0.6.0 and Carthage 0.18.1.

[I've created a simple Xcode project](https://github.com/mdarnall/Simple-Xcode-Project) that targets swift 2.3 and Dobby 0.6.0 to illustrate the issue and attached it here along with Xcode build output. I'm going to see if there is something obvious to "fix" to solve it. 


[carthage-xcodebuild.imDO6H copy.txt](https://github.com/trivago/Dobby/files/730703/carthage-xcodebuild.imDO6H.copy.txt)


[Simple-Xcode-Project-master.zip](https://github.com/trivago/Dobby/files/730694/Simple-Xcode-Project-master.zip)

The issue can also just be reproduced by checking out `0.6.0` and running `carthage build --no-skip-current --platform iOS`

**Comment by mdarnall** [comment]
Ok here is what I found In `0.6.0` where `SWIFT_VERSION = 2.3`: 

Xcode will fail to build Dobby for a device (`xcodebuild -sdk iphoneos`  or "Generic iOS Device in Xcode") because it links against XCTest which seems to have been built without bitcode in the Swift 2.3 toolchain. 

```
Auto-Linking supplied '/Applications/Xcode.app/Contents/Developer/Toolchains/Swift_2.3.xctoolchain/usr/lib/swift/iphoneos/libswiftXCTest.dylib', '/Applications/Xcode.app/Contents/Developer/Toolchains/Swift_2.3.xctoolchain/usr/lib/swift/iphoneos/libswiftXCTest.dylib' does not contain bitcode. You must rebuild it with bitcode enabled (Xcode setting ENABLE_BITCODE), obtain an updated library from the vendor, or disable bitcode for this target.
```

So I believe the "right" fix for `0.6.0` is to explicitly disable bitcode from the `Dobby-iOS` target all together. By doing so xcodebuild/Xcode/carthage can successfully build a framework targeted for a physical device. 

Here is the change: https://github.com/trivago/Dobby/compare/0.6.0...mdarnall:no-bitcode-swift23

I'd think this would need to go into a patched `0.6.1` branch/tag/release.

**Comment by felixvisee** [comment]
Fixed by #42.

**Comment by felixvisee** [comment]
Released as `0.6.2` (it was supposed to be `0.6.1`, but I rushed the tagging and forgot to actually bump the version). Anyways 🚀
